public/app/plugins/datasource/prometheus/result_transformer.tsx

```diff
@@ -127,21 +133,38 @@ function getText(
   seriesLabels: PromMetric,
   exemplarTraceIDDestination?: ExemplarTraceIDDestination
 ) {
-  let traceID = exemplar.labels.traceID;
+  let traceIDComponent: any;
+  if (exemplarTraceIDDestination) {
+    const traceID = exemplar.labels[exemplarTraceIDDestination.name];
+    const tag = ReactDOMServer.renderToString(<Tag name={traceID} colorIndex={6} />);
+    const href = exemplarTraceIDDestination.url.replace('${value}', traceID);
+    const anchorElement = `<a href="${href}" rel="noopener" target="_blank">${tag}</a>`;
+    traceIDComponent = anchorElement;
+  }
+
   const template = `
     <div>
-      <ul>
+      <div style="padding:10px">Series labels</div>
+      <table class="exemplars-table">
       ${Object.keys(seriesLabels)
-        .map(label => `<li>${label}: ${seriesLabels[label]}</li>`)
+        .map(label => `<tr><td>${label}</td><td>${seriesLabels[label]}</td></tr>`)
         .join('\n')}
-      </ul>
+      </table>
+      <div style="padding:10px">Exemplar</div>
+      <table class="exemplars-table">
+        ${Object.keys(exemplar.labels)
+          .map(label => {
+            if (label === exemplarTraceIDDestination?.name) {
+              return `<tr><td>${label}</td><td>${traceIDComponent}</td></tr>`;
+            }
+            return `<tr><td>${label}</td><td>${exemplar.labels[label]}</td></tr>`;
+          })
+          .join('\n')}
+        <tr><td>timestamp</td><td>${exemplar.timestamp}</td></tr>
+        <tr><td>Value</td><td>${exemplar.value}</td></tr>
+      </table>
     </div>`;
-  if (exemplarTraceIDDestination) {
-    traceID = exemplar.labels[exemplarTraceIDDestination.name];
-    const href = exemplarTraceIDDestination.url.replace('${value}', traceID);
-    const anchorElement = `<a href="${href}" rel="noopener" target="_blank">Go to ${traceID}</a>`;
-    return template + anchorElement;
-  }
+
```

Comment:
This feels wonky, I thought this exemplar data would be part of the DataFrame a the trace id link be done via data link. So all this table rendering could be done by Grafana and shared between all data sources. But understand that this is just for a demo

Reply:
Yep I think that is the future for this that I only need to pass the DataFrame with datalink and I don't have to pass an html like this.